hackathon_site/dashboard/frontend/src/slices/hardware/hardwareSlice.ts

```diff
@@ -190,13 +225,35 @@ const hardwareSlice = createSlice({
             state.isMoreLoading = false;
             state.error = payload?.message || "Something went wrong";
         });
+
+        builder.addCase(getUpdatedHardwareDetails.pending, (state) => {
+            state.isUpdateDetailsLoading = true;
+            state.error = null;
+        });
+
+        builder.addCase(getUpdatedHardwareDetails.fulfilled, (state, { payload }) => {
+            if (payload) {
+                state.isUpdateDetailsLoading = false;
+                state.error = null;
+                state.hardwareIdInProductOverview = payload.id;
+                hardwareAdapter.updateOne(state, {
+                    id: payload.id,
+                    changes: payload,
+                });
+            }
+        });
+
+        builder.addCase(getUpdatedHardwareDetails.rejected, (state, { payload }) => {
```

Comment:
@KarandeepLubana this PR looks almost done, the only thing here is you update the error state but you don't display the error anywhere. Also, this error state is for loading hardware on the inventory page so you should create a new error just for loading the hardware details